spring-cloud-zookeeper-discovery/src/main/java/org/springframework/cloud/zookeeper/discovery/ZookeeperDiscoveryProperties.java

```diff
@@ -70,7 +79,8 @@ private ZookeeperDiscoveryProperties() {}
 
 	public ZookeeperDiscoveryProperties(InetUtils inetUtils) {
 		this.hostInfo = inetUtils.findFirstNonLoopbackHostInfo();
-		this.instanceHost = this.hostInfo.getIpAddress();
+		this.instanceHost = this.hostInfo.getHostname();
+		this.instanceIpAddress = this.hostInfo.getIpAddress();
```

Comment:
Maybe we could throw `IllegalStateException` if you set `preferIpAddress` and don't have `instanceIpAddress` set? cc @spencergibb